### ISSUE TITLE
 Clarification on Parameter Settings for Training from Scratch for Cell-type Annotation

### ISSUE THREAD

**Issue by qiliu-ghddi** [bowang-lab/scGPT#146]
Hi, thank you for your interest! The training from scratch can also be readily supported by the same tutorial [notebook](https://github.com/bowang-lab/scGPT/blob/main/tutorials/Tutorial_Annotation.ipynb). 

In the following cell in the notebook:

![image](https://github.com/bowang-lab/scGPT/assets/11674033/015bb881-d141-4e56-9d18-39251213fd4b)

Set `load_model=None` then the model loading step would be skipped. Note that we did find it easier to train a smaller model from scratch rather than a large one. For this reason, the default model layer_size and nlayers are set to small values in the screenshot. You may simply follow these settings.

_Originally posted by @subercui in https://github.com/bowang-lab/scGPT/issues/51#issuecomment-1661224239_

**Comment by qiliu-ghddi** [comment]
Hello, we are attempting to replicate the training of scGPT from scratch for Cell-type Annotation. In our effort to train the model from scratch, I have set the `load_model` parameter to `None` and have reduced both the `layer_size` and `nlayers`. but the best results on the Multiple Sclerosis dataset we got is `Accuracy: 0.592, Precision: 0.381, Recall: 0.370, Macro F1: 0.328`.
which had a significant discrepancy in results compared to those reported. 

![Snipaste_2024-01-12_15-33-25](https://github.com/bowang-lab/scGPT/assets/87366652/21cb90a3-2acc-4361-bc0b-1328713a4c35)

![Snipaste_2024-01-12_15-38-46](https://github.com/bowang-lab/scGPT/assets/87366652/3984973a-192f-424e-a18b-31de6eaa7ccb)


Could you please provide us with the exact parameter settings you used in the paper for training the model from scratch? Details on the architecture, learning rate, number of epochs, and any other relevant hyperparameters would be highly beneficial.

![Snipaste_2024-01-12_15-29-51](https://github.com/bowang-lab/scGPT/assets/87366652/6565e6e5-f740-4af8-b115-1e0dfbd5d44c)

This information will greatly aid in understanding the nuances of the training process and improve the reproducibility of the results. Thank you for your time and assistance.

**Comment by subercui** [comment]
Hi, thank you for the question. This is likely due to the recent code base updates, which are primarily designed for fine-tuning and zero-shot settings. I will find out our previous runs first and share them with you. BTW, just to confirm, you did set the model to a smaller size, such as the ones in the tutorial default params.

**Comment by subercui** [comment]
By the way, we have corrected this section in our recent manuscript which explicitly states the from-scratch comparison is on a smaller model. We haven't released the new manuscript online but will likely soon.

**Comment by subercui** [comment]
Here is one run I found, https://wandb.ai/scformer-team/scGPT-public/runs/2gf70rfv/logs . The logs and code file in wandb may be helpful. Please note that this was run using legacy code, so the variable and class names can be different. Will try to provide a newer version if needed. In the meantime, I would strongly recommend using the fine-tuning setting if possible, it can usually achieve more robust and better performance.
